### PR TITLE
GitHub oauth authentication

### LINKED ISSUE #10: User Dashboard: Github OAuth
**Details:** We want a user to have the option to connect their Github account and retrieve a user specific token automatically.

```
As a user
When I visit /dashboard
Then I should see a link that is styled like a button that says "Connect to Github"
And when I click on "Connect to Github"
Then I should go through the OAuth process
And I should be redirected to /dashboard
And I should see all of the content from the previous Github stories (repos, followers, and following)
```

**Tips:** Testing this is tricky. Don't worry about testing during the spike. In fact, TDD'ing this is really tricky if you've never done it before. Get a working implementation first even when you are pairing and backfill a test. You *should not* try to fill out the form and click buttons on Github in your tests. Omniauth gives us a special way to test this. Use these docs as a starting point: https://github.com/omniauth/omniauth/wiki/Integration-Testing

- [x] Uses Omniauth Github (does not hand roll the OAuth handshake)
- [ ] Tests the handshake using Omniauth Mock Auth (see above tips section).

### PR DESCRIPTION
**Pull request by tony-gomes**
Add Connect Github link to dashboard
Add omniauth route to githubs#create action
Add Githubs Controller for oauth w/ private params method

### CODE REVIEW

File: app/controllers/githubs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    response = request.env['omniauth.auth']
+    current_user.update(token: github_params(response))
+    redirect_to dashboard_path
+  end
+
+  private
+
+  def github_params(response)
+    response["credentials"]["token"]
+  end
```
Comment (javier-aguilar) [inline app/controllers/githubs_controller.rb:12]: Regarding the github_params and line 4. I wrote something similar with tutorials create
`tutorial = Tutorial.create(tutorial_info_params(tutorial_info))`
``` 
def tutorial_info_params(tutorial)
    { title: tutorial.title,
      description: tutorial.description,
      thumbnail: tutorial.thumbnail,
      playlist_id: tutorial.id }
  end
```

**Comment by javier-aguilar** [comment]
Everything looks solid. It looks like the Travis builds are failing on one of the tests. It seems that we are checking for the "Github" section but in this code, we introduce "Connect to Github" which causes the test to fail. All that should be needed is wrapping that expect test in a 'within'.

**Comment by javier-aguilar** [comment]
Also, I didn't see any tests. I understand Omniauth Mock Auth is something we should when testing.

File: app/controllers/githubs_controller.rb (hunk 1 above)
Comment (tony-gomes) [reply app/controllers/githubs_controller.rb:12]: So you want this syntax for the params?

**Comment by tony-gomes** [comment]
> Everything looks solid. It looks like the Travis builds are failing on one of the tests. It seems that we are checking for the "Github" section but in this code, we introduce "Connect to Github" which causes the test to fail. All that should be needed is wrapping that expect test in a 'within'.

A within would require some sort of target but none of the sections or other title headings have classes. So I added a css heading target to the expect statement & it passes now.

**Comment by tony-gomes** [comment]
> Also, I didn't see any tests. I understand Omniauth Mock Auth is something we should when testing.

Based on my understanding of the user story - tests were not required for this "spike"...
"...Don't worry about testing during the spike..."

But from what I heard, we will be going over testing oauth during Monday lessons. Can revisit my branch if this is the case. Otherwise, I can revisit as a refactor after the rest is finished since testing is explicitly not requested here but just alluded to...

**Pull request merged**